### PR TITLE
feat(bmc): New github wrapper to use github api by pat. BM-878

### PR DESCRIPTION
**Pull request by Wentao-Kuang**
#### Description

Remove all the github commands that using ssh key and update to use github API by PAT.

#### Intention

When running cli we can directly create pull requests by using the github API.

#### Checklist
*If not applicable, provide explanation of why.*
- [ ] Tests updated
- [ ] Docs updated
- [ ] Issue linked in Title

### CODE REVIEW

File: src/commands/basemaps-github/__test__/create-manifest.test.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { describe, it } from 'node:test';
+
+import prettier from 'prettier';
+
+import { DefaultPrettierFormat } from '../make.cog.github.js';
+
+describe('DefaultPrettierFormat', () => {
+  it('should be the same prettier format with @linzjs/style', async () => {
+    const cfg = await prettier.resolveConfigFile('@linzjs/style/.prettierrc.cjs');
+    if (cfg == null) assert.ok(cfg !== null, 'Failed to read @linzjs/style/.prettierrc.cjs');
```
Comment (blacha) [inline src/commands/basemaps-github/__test__/create-manifest.test.ts:11]:
```suggestion
    assert.ok(cfg, 'Failed to read @linzjs/style/.prettierrc.cjs');
```

File: src/commands/basemaps-github/__test__/create-manifest.test.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    const options = await prettier.resolveConfig(cfg);
+    if (options == null) assert.ok(options !== null, 'Failed to resolve prettier config');
```
Comment (blacha) [inline src/commands/basemaps-github/__test__/create-manifest.test.ts:13]:
```suggestion
    assert.ok(options, 'Failed to resolve prettier config');
```

File: src/commands/basemaps-github/__test__/create-manifest.test.ts (hunk 1 above)
Comment (Wentao-Kuang) [reply src/commands/basemaps-github/__test__/create-manifest.test.ts]: Nice, I don't know that asser.ok can automatically check non exist. I just fixed them.

File: src/commands/basemaps-github/github.ts [hunk 3]
```diff
@@ -2,115 +2,220 @@ import { Env, LogType } from '@basemaps/shared';
...
+    if (owner == null || repo == null) throw new Error(`Badly formatted repository name: ${repository}`);
+    this.owner = owner;
     this.repo = repo;
-    this.logger = logger;
-    this.org = org;
-    this.repoName = repoName;
 
     const token = Env.get(Env.GitHubToken);
-    if (token == null) throw new Error(`Please set up ${Env.GitHubToken} environment variable.`);
+    if (token == null) throw new Error('Please set up github token environment variable.');
```
Comment (blacha) [inline src/commands/basemaps-github/github.ts:31]: how come we got rid of the $ env var name?

File: src/commands/basemaps-github/github.ts [hunk 4]
```diff
@@ -2,115 +2,220 @@ import { Env, LogType } from '@basemaps/shared';
...
+   * Create a blob object in git
    */
-  configUser(): void {
-    const email = Env.get('GIT_USER_EMAIL') ?? 'basemaps@linz.govt.nz';
-    const name = Env.get('GIT_USER_NAME') ?? 'basemaps[bot]';
-    this.logger.info({ repository: this.repo }, 'GitHub: Config User Email');
-    execFileSync('git', ['config', 'user.email', email], { cwd: this.repoName }).toString().trim();
-    this.logger.info({ repository: this.repo }, 'GitHub: Config User Name');
-    execFileSync('git', ['config', 'user.name', name], { cwd: this.repoName }).toString().trim();
+  async createBlobs(content: string, path: string, logger: LogType): Promise<Blob> {
```
Comment (blacha) [inline src/commands/basemaps-github/github.ts:85]: blob"s" implies multiple files?

File: src/commands/basemaps-github/github.ts [hunk 5]
```diff
@@ -2,115 +2,220 @@ import { Env, LogType } from '@basemaps/shared';
...
     return response.data.number;
   }
 }
+
+export interface GithubFiles {
+  path: string;
+  content: string;
+}
+
+export class GitHubCreatePR {
```
Comment (blacha) [inline src/commands/basemaps-github/github.ts:180]: This doesnt need to be a class, could just make it a function, saves you doing `new GithubFileS(xxx).createPR()`

File: src/commands/basemaps-github/github.ts [hunk 6]
```diff
@@ -2,115 +2,220 @@ import { Env, LogType } from '@basemaps/shared';
...
-    for (const path of paths) {
-      execFileSync('git', ['add', path], { cwd: this.repoName }).toString().trim();
+  async getContent(path: string, logger: LogType): Promise<string> {
+    logger.info({ path }, 'GitHub API: Get Content');
+    const response = await this.octokit.rest.repos.getContent({ owner: this.owner, repo: this.repo, path });
+    if (!this.isOk(response.status)) throw new Error('Failed to get aerial TileSet config.');
+    if ('content' in response.data) {
+      return Buffer.from(response.data.content, 'base64').toString();
+    } else {
+      throw new Error('Unable to find the content.');
```
Comment (blacha) [inline src/commands/basemaps-github/github.ts:110]: its generally a good idea to put context into errors like the path and repo ?

**Review by blacha — APPROVED** [review]

**Pull request merged**